docs/basic-component.md

```diff
@@ -0,0 +1,247 @@
+# Basic component recipe
+
+Recipe for creating a basic component.
+
+- [Component structure](#component-structure)
+  - [Subcomponents](#subcomponents)
+  - [Hooks and Utils](#hooks-and-utils)
+  - [index.ts](#indexts-1)
+  - [Component.tsx](#componenttsx)
+  - [Styles](#styles)
+  - [Tests](#tests)
+- [Component.tsx template](#componenttsx-template)
+  - [Props](#props)
+  - [The component](#the-component)
+  - [Display name](#display-name)
+
+## Component structure
+
+Given the component `PancakeStack`, the component structure will follow this:
+
+```
+/* Component */
+src/
+  PancakeStack/
+    components/
+      - Pancake/
+        - index.ts
+        - Pancake.tsx
+        - Pancake.scss
+        - Pancake.spec.tsx
+      - Topping/
+        - index.ts
+        - Topping.tsx
+        - Topping.scss
+        - Topping.spec.tsx
+    hooks/
+      - index.ts
+      - useHook.ts
+      - useHook.spec.ts
+    utils/
+      - index.ts
+      - groupedFunctions.ts (eg. getters)
+      - groupedFunctions.spec.ts
+      - functionName.ts
+      - functionName.spec.ts
+    - index.ts
+    - PancakeStack.tsx
+    - PancakeStack.scss
+    - PancakeStack.spec.tsx
+```
+
+### Subcomponents
+
+Subcomponents live in the `components/` directory and should not have subcomponents of their own. If you are finding that you want subcomponents for your subcomponents, then one (or both) should be a component instead.
+
+Aside from the above, subcomponents should follow the same structure as a normal component.
+
+Subcomponents living within their own directories have the benefits of keeping files small and clean, and allows for mocking for tests.
+
+### Hooks and Utils
+
+Hooks and Utils go into their own respective folders, with their filename reflective of either the function name or a collective name based on the functionality of functions within the file.
```

Comment:
nit: I don't think we need to capitalise utils here

Reply:
Fixed :)